### PR TITLE
Optimize args.pop()/push() as args[args.length - 1]

### PR DESCRIPTION
**Pull request by chinesedfan**
#28.

### CODE REVIEW

**Comment by TheLarkInn** [comment]
@chinesedfan So sorry we've missed this the past x months. I'm going to test this locally and verify that this doesn't cause any issues. cc @sokra

File: lib/Tapable.js [hunk 1]
```diff
@@ -128,19 +128,19 @@ Tapable.prototype.applyPluginsBailResult1 = function applyPluginsBailResult1(nam
 
 Tapable.prototype.applyPluginsAsyncSeries = Tapable.prototype.applyPluginsAsync = function applyPluginsAsyncSeries(name) {
 	var args = Array.prototype.slice.call(arguments, 1);
-	var callback = args.pop();
+	var callback = args[args.length - 1];
```
Comment (mikesherov) [inline lib/Tapable.js]: this is broken. `pop` removes elements from the array.

Comment (chinesedfan) [inline lib/Tapable.js]: Not really. Original codes remove the last element here, but push a new one after a while. Replacing the last element directly will be faster, without breaking changes. Please refer to the original implementation of `applyPluginsParallelBailResult`.

Comment (mikesherov) [inline lib/Tapable.js]: I see. Thanks for correcting me.

**Comment by sokra** [comment]
new major version doesn't use arguments anymore

**Pull request closed without merging**